### ISSUE TITLE
Obsolete-Attribute with interpolated string as message does not raise warnings

### ISSUE THREAD

**Issue by dd-inno** [dotnet/roslyn#64605]
**Version Used**: 
Microsoft Visual Studio Professional 2022 (64-bit) - Current
Version 17.3.3

Microsoft (R) Visual C# Compiler Version 4.3.0-3.22423.10 (b35bb0ba)

**Steps to Reproduce**:

[sharplab repro](https://sharplab.io/#gist:ad48e238601ae6acdb1a3f84b71522b6)

```csharp
using System;

internal class Program
{
    static void Main(string[] args)
    {
        // this does not raise a warning
        var object1 = new LegacyObject1();

        // but this does
        var object2 = new LegacyObject2();
    }
}

[Obsolete($"Do not use {nameof(LegacyObject1)}")]
public class LegacyObject1
{
}

[Obsolete("Do not use" + nameof(LegacyObject2))]
public class LegacyObject2
{
}
```

**Expected Behavior**:
I'm not sure about the actual problem. Either the attribute for `LegacyObject1` should not compile, as the value for `message` is not compile-time constant (or is it?), or the instantiation of it should raise a CS0618 warning.

**Actual Behavior**:
This compiles and does not raise a warning.

**Comment by Youssef1313** [comment]
I took a very quick look:

https://github.com/dotnet/roslyn/blob/91a2af3e4dfb529417bd3c8f0131b43a1a5895ca/src/Compilers/CSharp/Portable/Symbols/Symbol_Attributes.cs#L177-L192

In this code path, we end up with `attributeData` with `HasErrors` being true, so we branch into the `else` with `obsoleteData = null`

**Comment by Youssef1313** [comment]
It looks like the binder used for early well-known attribute decoding is unable to bind interpolated strings:

![image](https://user-images.githubusercontent.com/31348972/194941564-c444ea19-30f0-40c0-9bb9-2687fcc1f672.png)

**Comment by dd-inno** [comment]
Thank you for analysing and fixing this so quickly!
